source/utils/Helper.cpp

```diff
@@ -262,23 +262,42 @@ namespace Helper
     {
         return TTostring(d);
     }
+    std::string toString(int64_t d)
+    {
+        return TTostring(d);
+    }
+    #ifndef __OpenBSD__
```

Comment:
Better move `toString(size_t d)` undef `#ifdef __OpenBSD__` then drop `#ifdef __LP64__`.

Reply:
I propose we merge your PR then will apply your changes to this one keeping the override fixes ... What do you think ?

Reply:
Sorry, I had a typo in my comment (edited). OpenBSD probably only needs `toString(size_t d)` only on 64bit platforms.

Reply:
If my #1286 introduces a regression on OpenBSD then it shouldn't be merged.

Reply:
Yes ideally the toString might be better as generics but that would require too much changes I fear.

Reply:
I ve tested my branch on FreeBSD and OpenBSD 64 bits so far by the way.

Reply:
i386 or x86 (32bit) is not dead yet. It has Tier1 support on FreeBSD.
```c++
source/gamemap/TileContainer.cpp:400:52: error: call to 'toString' is ambiguous
            OD_LOG_ERR("Unknown neighbor index=" + Helper::toString(i));
                                                   ^~~~~~~~~~~~~~~~
source/utils/LogManager.h:32:155: note: expanded from macro 'OD_LOG_ERR'
#define OD_LOG_ERR(_message)                      LogManager::getSingleton().logMessage(LogMessageLevel::CRITICAL, __FILE__, __LINE__, (std::string("") + _message))
                                                                                                                                                          ^~~~~~~~
source/utils/Helper.h:131:17: note: candidate function
    std::string toString(float f, unsigned short precision = 6);
                ^
source/utils/Helper.h:132:17: note: candidate function
    std::string toString(double d, unsigned short precision = 6);
                ^
source/utils/Helper.h:133:17: note: candidate function
    std::string toString(int8_t d);
                ^
source/utils/Helper.h:134:17: note: candidate function
    std::string toString(uint8_t d);
                ^
source/utils/Helper.h:135:17: note: candidate function
    std::string toString(int16_t d);
                ^
source/utils/Helper.h:136:17: note: candidate function
    std::string toString(uint16_t d);
                ^
source/utils/Helper.h:137:17: note: candidate function
    std::string toString(int32_t d);
                ^
source/utils/Helper.h:138:17: note: candidate function
    std::string toString(int64_t d);
                ^
source/utils/Helper.h:144:17: note: candidate function
    std::string toString(uint64_t d);
                ^
```

Reply:
Ok so since your PR for Helpers change seem to work (tested locally your changes with my own branch) might be good to merge yours then will rebase this one. What do you think ?

Reply:
If the changes here won't touch `toString` helpers simply rebase against  `development` branch. No need to wait for #1286 to be merged. Obviously, history would look better if you squash reverts, split unrelated bits into separate commits and document compiler error messages.

Reply:
OK will try later in the day. Cheers.